include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -387,8 +389,9 @@ inline void multiplexer::do_start_receive()
 inline void multiplexer::discard_message() {
     boost::system::error_code error;
     endpoint_type remote_endpoint;
+    static char empty_buffer_storage;
     next_layer().receive_from
-        ( boost::asio::buffer(static_cast<char*>(nullptr), 0)
+        ( boost::asio::buffer(&empty_buffer_storage, 1)
```

Comment:
@inetic As I mentioned by email, I think the fact that zero length UDP datagrams work at all on any platform is undefined behaviour - I can find explicitly documented support for Linux and Windows, but not FreeBSD nor OS X nor POSIX. Also, jumbo UDP datagrams explicitly set the size to zero to indicate a following jumbogram, so what works now may not work in a world with jumbo datagram support. Finally, we definitely know that zero sized UDP datagrams don't work on OS X from https://github.com/joyent/node/issues/8023.